Add `periodic_callable`

basnijholt opened the pull request:
@jbweston, what about this? Closes #206.

I haven't really tested it but I suppose you can first see it this matches your use-case.

edit: I tested it and now it works

basnijholt committed:
Add `periodic_callable`

basnijholt committed:
Fix serdes

basnijholt committed:
pass through

basnijholt committed:
order

basnijholt committed:
fix order args

basnijholt committed:
No serialized args in env

basnijholt committed:
arg order

basnijholt committed:
clean

basnijholt committed:
consistent kwargs order

jbweston commented:
Great! The only change I would add would be to pass the learner's "fname" along with the learner itself (like 'to_dataframe' takes)

jbweston committed:
Make 'periodic_callable' expect a function of 2 arguments.

The first argument is the 'fname' of the learner, and the
second argument is the learner itself. The fname is needed
to use as a base, if the periodic callable will save some
additional data.

jbweston committed:
Fix syntax error

basnijholt reviewed (commented):
Looks good @jbweston! Go ahead and merge if you are happy with the implementation and it works for your use-case.
